prince-of-versions/src/test/java/co/infinum/princeofversions/JsonConfigurationParserTest.java

```diff
@@ -148,7 +161,7 @@ public void validUpdateFullWithSdkValuesJson() throws Throwable {
         PrinceOfVersionsConfig config = parser.parse(ResourceUtils.readFromFile("valid_update_full_with_sdk_values.json"));
         assertThat(
             config
-        ).isEqualTo(
+        ).isEqualToComparingFieldByField(
```

Comment:
What does this change do?

Reply:
I noticed that the test would go through while completely disregarding the `requirements`. You could write `withRequirements(MapUtil.from(entry("whatever", "whatever")))` & it would still pass the test. Comparing it field by field solves the issue & makes sure all of the values are equal.
I did mean to ask though, what is the reasoning behind the first one failing (why the acceptable criteria for passing is just having a map & not its contents)?

Reply:
That's because equals method in PrinceOfVersionsConfig don't compare requirements and that is bug.